test/client/fixtures/sandbox/iframe-test.js

```diff
@@ -294,6 +294,19 @@ test("'body.appendChild' method works incorrectly in the particular case (GH-421
     checkIframes();
 });
 
+test('hammerhead must be initialized after the document with same-domain src is rewrited (GH-1325)', function () {
+    return createTestIframe({ src: getSameDomainPageUrl('../../data/iframe/simple-iframe.html') })
+        .then(function (iframe) {
+            var stringifiedNativeWriteFn = nativeMethods.documentWrite.toString();
```

Comment:
Let's check that hammerhead is properly initialized more strictly:
- set src attribute for link and check proxied url
- perform `document.write`
- again set src attribute for link and check proxied url